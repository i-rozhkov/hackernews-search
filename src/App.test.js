import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App, { Search, Button, Table } from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has correct snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Search', () => {
  it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search onChange={() => {}} onSubmit={() => {}}>Поиск</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has correct snapshot', () => {
    const component = renderer.create(
      <Search onChange={() => {}} onSubmit={() => {}}>Поиск</Search>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button', () => {
  it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button onClick={() => {}}>Дай мне больше</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has correct snapshot', () => {
    const component = renderer.create(
      <Button onClick={() => {}}>Дай мне больше</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Table', () => {
  const props = {
    list: [
      { title: '1', author: '1', num_comments: '1', points: 2, objectID: 'y'},
      { title: '2', author: '2', num_comments: '2', points: 2, objectID: 'z'},
    ],
    sortKey: 'TITLE',
    isSortReverse: false,
  }

  it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table {...props} onClick={() => {}} onDismiss={() => {}} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has correct snapshot', () => {
    const component = renderer.create(
      <Table {...props} onClick={() => {}} onDismiss={() => {}} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows two items in list', () => {
    const element = shallow(<Table {...props} onClick={() => {}} onDismiss={() => {}} />);
    expect(element.find('.table-row').length).toBe(2);
  });
});