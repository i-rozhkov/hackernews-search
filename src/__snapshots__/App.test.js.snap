// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has correct snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Поиск
      </button>
    </form>
  </div>
  <div
    className="table"
  >
    <div
      className="table-header"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Заголовок
        </button>
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Автор
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Комментарии
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Очки
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        Архив
      </span>
    </div>
  </div>
  <div
    className="interactions"
  >
    <div>
      Загрузка...
    </div>
  </div>
</div>
`;

exports[`Button has correct snapshot 1`] = `
<button
  className={undefined}
  onClick={[Function]}
  type="button"
>
  Дай мне больше
</button>
`;

exports[`Search has correct snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value={undefined}
  />
  <button
    type="submit"
  >
    Поиск
  </button>
</form>
`;

exports[`Table has correct snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-header"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <button
        className="button-inline button-active"
        onClick={[Function]}
        type="button"
      >
        Заголовок
      </button>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Автор
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Комментарии
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Очки
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      Архив
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href={undefined}
      >
        1
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Отбросить
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href={undefined}
      >
        2
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Отбросить
      </button>
    </span>
  </div>
</div>
`;
